### PR TITLE
[Debugger Plugin]: Continue to condition & misc. improvements

### PR DESCRIPTION
**Pull request by caisq**
Main new feature:
* Ability to continue to a given condition for any of the watched
  tensors. This is presented as a new section in the pop-up dialog
  of the "Continue..." button, while the existing continue n times
  of Session.run is put into the first section of the dialog.
* The new continue-to momde supports the following modes:
  * Contains inf or nan
  * Contains inf
  * Contains nan
  * Max > or < ref value
  * Min > or < ref value
  * Range (Max - Min) > or < ref value
  * Mean > or < ref value
  * Stddev > or < ref value
* The pop-up dialog for Continue... is refactored into its own
  Polymer component: tf-debugger-continue-to-dialog.html

Misc. improvements:
* Fix a bug wherein continue across multiple Session.run keys does
  not render graph for new run key.
* Move the toast from the bottom left to the bottom center. This
  prevents the toast from obscuring the Step and Continue... buttons.

### CODE REVIEW

**Comment by caisq** [comment]
Example screenshot:
health_pills_3

File: tensorboard/plugins/debugger/tf_debugger_dashboard/health-pills.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+export type TensoConditionPredicate =
+    (healthPill: number[], refValue?: number) => boolean;
+
+
+export interface TensorCondition {
+  description: string;
+  predicate: TensoConditionPredicate;
+}
+
+function isHealthPillUninitizliedOrUnsupported(healthPill: number[]): boolean {
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/health-pills.ts]: nit: spelling of initialized

File: tensorboard/plugins/debugger/tf_debugger_dashboard/health-pills.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+ * Any condition value that involves numerical comparison require a value
+ *   in the `refValue` argument (see below).
+ * @param healthPill The health pill.
+ * @param refValue Threshold required by some of the condition values. If not
+ *   provided for such conditions, an Error will be thrown.
+ * @return Whether the condition is met.
+ * @throws Error on in valid condition values or missing refValue for conditions
+ *   that requires it.
+ */
+export type TensoConditionPredicate =
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/health-pills.ts]: nit: spelling of Tensor

File: tensorboard/plugins/debugger/tf_debugger_dashboard/health-pills.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    description: 'Standard deviation <',
+    predicate: (healthPill: number[], refValue?: number) => {
+      checkRefValue(refValue, 'STDDEV_LT');
+      return Math.sqrt(healthPill[11]) < refValue;
+    },
+  },
+};
+
+/**
+ * Convert the human-readable description of a tensor-value conditio to its key.
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/health-pills.ts]: nit: spelling of condition

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-continue-dialog.html [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+          return;
+        }
+        const tensorConditionKey =
+            tf_debugger_dashboard.tensorConditionDescription2Key(this._selectedTensorCondition);
+        if (tensorConditionKey == null) {
+          console.error(
+              'Invalid Tensor Condition name:' + this._selectedTensorCondition);
+        }
+        // Validate that this._tensorConditionRefValue is a number.
+        if (isNaN(this._tensorConditionRefValue)) {
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-continue-dialog.html]: You can use `if !(_.isFinite(this._tensorConditionRefValue))`. That also checks for no Infinity.

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-dashboard.html [hunk 5]
```diff
@@ -646,36 +654,65 @@ <h2>Continue...</h2>
                 'Invalid long-polling response type: ', responseType);
           }
 
+          if (this._continueToType != null) {
+            this._processContinueTo(responseType, responseData);
+          }
+
+          // Long polling loop: initialite the next long polling.
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-dashboard.html]: nit: spelling of initialize

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-continue-dialog.html [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+        <paper-icon-button
+          class="go-button"
+          icon="arrow-forward"
+          title="Session Runs Go"
+          on-tap="_sessionRunGoButtonCallback">
+        </paper-icon-button>
+      </div>
+      <div class="continue-to-type">
+        <div class="continue-to-type-name">
+          Till Condition Met by Watched Tensor
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-continue-dialog.html]: s/Tensor/Tensors

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-continue-dialog.html [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+      open() {
+        this.$.continueDialog.open();
+      },
+
+      close() {
+        this.$.continueDialog.close();
+      },
+
+      _sessionRunGoButtonCallback() {
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-continue-dialog.html]: This is a little tedious, but could you add a little documentation on when these callbacks would run?

**Comment by caisq** [comment]
Thanks for the review!

**Pull request merged**